solvers/noncvx_pro.py

```diff
@@ -7,29 +7,49 @@
     from numpy.linalg import norm
     import scipy.optimize as sciop
     from scipy.sparse import issparse
+    from sklearn.linear_model._base import _preprocess_data
 
 
 class Solver(BaseSolver):
     name = "noncvx-pro"
 
     stopping_strategy = 'iteration'
     stopping_criterion = SufficientDescentCriterion(eps=1e-10, patience=5)
+    requirements = ["scikit-learn"]
```

Comment:
I'd avoid having sklearn as a requirement for this solver (if we do it manually as proposed above, this requirement is no longer needed)

Reply:
Handled in 2b634fcd7d297676fe9f00bf48360966ad813bef